### PR TITLE
[WC-186, WC-205] Implement Course Video Service and fix Some Change JWT, Schema for Chef service

### PR DESCRIPTION
**Pull request by Billy19191**
# Description

[//]: # 'Please include a summary of the changes and the related issue. (You can write below without deleting this comment)'

### What

[//]: # 'Briefly describe this feature / fix / refactor (You can write below without deleting this comment)'
Implement video service endpoint and allow user and chef to create,get, update about course. Apart from that, also fix some stuff of chef service about JWT / authorization.

### Why

[//]: # 'Describe the reason why we need this feature / fix / refactor or refer to Jira Ticket (You can write below without deleting this comment)'
Allow user to enroll course, get their learning courses, whereas, for chef, allow to create, updates new courses.

## Type of change

Please check the relevent option with `x`.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if needed)
- [x] My changes generate no new warnings
- [x] I have tested on my `Local Machine`

## Release Notes

- **New Features**
  - Introduced JWT authentication capabilities across the application.
  - Added new methods for course enrollment and video retrieval in the user service.
  - Enhanced the user controller with endpoints for managing course videos and enrollments.
  - Added request body DTOs for chef sign-up, login, profile updates, and course management.
  - Introduced a new video service with capabilities for handling course videos.
  - Added new classes for handling event messages related to chefs and courses.

- **Bug Fixes**
  - Improved error handling across multiple services and controllers.

- **Documentation**
  - Added a comprehensive README for the video service, detailing setup and usage.

- **Refactor**
  - Streamlined parameter handling in various methods for consistency.
  - Restructured modules and services to improve maintainability.

- **Chores**
  - Updated dependencies across services to enhance functionality and security.
  - Introduced ESLint and Prettier configurations for code quality.

### CODE REVIEW

**Comment by Billy19191** [comment]
@AungKhantThaw47 Please ping me discord when you're reviewed.

File: chef-service/src/app.controller.ts [hunk 1]
```diff
@@ -5,11 +5,10 @@ import { EventPattern } from '@nestjs/microservices';
 @Controller()
 export class AppController {
   constructor(private readonly appService: AppService) {}
-  @EventPattern('chef')
   @EventPattern('chef/signup')
-  async postSignUpChef(@Body() body: any[]) {
+  async postSignUpChef(payload): Promise<any> {
```
Comment (AungKhantThaw47) [inline chef-service/src/app.controller.ts]: can you use event message for this?

File: chef-service/src/app.controller.ts [hunk 2]
```diff
@@ -47,21 +46,21 @@ export class AppController {
   }
 
   @EventPattern('chef/updateProfile')
-  async updateProfileChef(@Body() body: any): Promise<any> {
+  async updateProfileChef(payload: any): Promise<any> {
```
Comment (AungKhantThaw47) [inline chef-service/src/app.controller.ts]: like mentioned above

File: chef-service/src/app.service.ts [hunk 3]
```diff
@@ -167,18 +179,18 @@ export class AppService extends PrismaClient implements OnModuleInit {
     }
   }
 
-  async updateProfileChef(body: any): Promise<any> {
+  async updateProfileChef(payload: any): Promise<any> {
```
Comment (AungKhantThaw47) [inline chef-service/src/app.service.ts]: type "any"

**Review by AungKhantThaw47 — COMMENTED** [review]
Overall , I think that it is ok. For data type "any" , please change class type if you have time.

**Comment by Billy19191** [comment]
> Overall , I think that it is ok. For data type "any" , please change class type if you have time.

Yes, for sure. I will try to finish this by the end of the day.

File: chef-service/src/app.controller.ts (hunk 1 above)
Comment (Billy19191) [reply chef-service/src/app.controller.ts]: Yes, will change for all

**Comment by Billy19191** [comment]
Yoooo @AungKhantThaw47, I push all code with class interface. Please review ;)

File: chef-service/src/app.controller.ts [hunk 4]
```diff
@@ -1,33 +1,40 @@
-import { Body, Controller } from '@nestjs/common';
+import { Controller } from '@nestjs/common';
 import { AppService } from './app.service';
 import { EventPattern } from '@nestjs/microservices';
+import {
```
Comment (AungKhantThaw47) [inline chef-service/src/app.controller.ts:4]: can combine as one import

File: chef-service/src/app.controller.ts [hunk 5]
```diff
@@ -1,33 +1,40 @@
...
+  ChefProfileUpdateEventMsg,
+  CourseUploadEventMsg,
+} from '../../lib/src/chef/event-msg.dto';
+import {
+  ChefLoginEventMsg,
+  ChefSignUpEventMsg,
+} from '../../lib/src/chef/event-msg.dto';
```
Comment (AungKhantThaw47) [inline chef-service/src/app.controller.ts:16]: use @lib

**Review by AungKhantThaw47 — COMMENTED** [review]
i think except response and some go to go

**Review by AungKhantThaw47 — APPROVED** [review]
green light

**Pull request merged**